azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -106,6 +106,7 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 			"overprovision": {
 				Type:     schema.TypeBool,
 				Optional: true,
+				Default:  true,
```

Comment:
bear in mind this will currently force a change for existing resources - so this is going to need to go out in a major version & we're going to want to call this out in the release notes (cc @katbyte)